### PR TITLE
Add NewRetryOptions() function

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -208,6 +208,11 @@ func (r RetryRule) WithBackoff(fn RetryBackoff) RetryRule {
 	return r
 }
 
+// NewRetryOptions returns the default retry options.
+func NewRetryOptions() RetryOptions {
```
Comment (elprans) [inline options.go:212]: Perhaps `NewDefaultRetryOptions`?

Comment (fmoor) [reply options.go:212]: We already have `NewTxOptions()` and `NewRetryRule()` that return the default value for their respective types.

Comment (elprans) [reply options.go:212]: So you are supposed to call `New..()` and then mutate properties directly?

Comment (fmoor) [reply options.go:212]: Sort of. It has a method chaining API.
```go
ruleA := edgedb.NewRetryRule()
ruleB := edgedb.NewRetryRule()
edgedb.NewRetryOptions().WithDefault(ruleA).WithCondition(NetworkError, ruleB)
```

**Review by elprans — APPROVED** [review]

**Pull request merged**